nbgrader/converters/base.py

```diff
@@ -359,3 +363,28 @@ def _handle_failure(gd):
 
             self.log.error(msg)
             raise NbGraderException(msg)
+
+    def recursively_archive(self, directory, archive_name):
+        all_files_and_folders = [ item for item in os.listdir(directory) if item != archive_name+".zip" ]
+
+        for item in all_files_and_folders:
+            full_path = os.path.join(directory, item)
+            if os.path.isfile(full_path):
+                self.archive.write(os.path.join(self.writer.build_directory, full_path.split(archive_name, 1)[-1][1:]), archive_name+"/"+full_path.split(archive_name, 1)[-1][1:])
+            elif os.path.isdir(full_path):
+                self.recursively_archive(full_path, archive_name)
+
+    def make_archive(self):
+        current_dir = self.writer.build_directory
+        archive_name = os.path.basename(self.writer.build_directory)
+
+        if archive_name+".zip" in os.listdir(current_dir):
+            os.remove(current_dir+"/"+archive_name+".zip")
```

Comment:
When working with paths, it is better to use https://docs.python.org/2/library/os.path.html#os.path.join
I am not sure that this concatenation works, for example, in windows (not sure if the package is intended to be run in windows machines, but that's beside my point).
Also, concatenations, in other contexts, are better handled by .format

Reply:
👍